src/components/relatedProducts/ComparisonModal.jsx

```diff
@@ -1,27 +1,51 @@
-import React, {useState, useEffect} from 'react'
+import React, { useState, useEffect } from 'react'
 import ReactDom from 'react-dom'
 import { useSelector, useDispatch } from 'react-redux';
-import {closeModal} from '../../redux/actions/index.js'
+import { closeModal } from '../../redux/actions/index.js'
 import api from '../../apis/RPandOL.js'
+import StarRatings from 'react-star-ratings'
 
-export default function ComparisonModal({onClose, id}) {
+export default function ComparisonModal({ onClose, id }) {
   const dispatch = useDispatch();
   const productId = useSelector((state) => state.productId);
-  const comparisonId = useSelector(state=>state.comparisonId);
+  const comparisonId = useSelector(state => state.comparisonId);
   const [baseDetails, setBaseDetails] = useState({});
   const [comparedDetails, setComparedDetails] = useState({});
-  let features = new Set([]);
+  const [baseStars, setBaseStars] = useState(0);
+  const [comparedStars, setComparedStars] = useState(0);
+  const [features, setFeatures] = useState([])
+  const [baseFeatures, setBaseFeatures] = useState([])
+  const [compFeatures, setCompFeatures] = useState([])
+
+  let featuresSet = new Set([]);
 
   const fetchData = async () => {
     try {
+      let bFeatures, cFeatures;
       const data = await api.getCardInfo(productId);
       const compared = await api.getCardInfo(comparisonId)
       setBaseDetails(data);
       setComparedDetails(compared);
-      console.log('basedetails are ', baseDetails)
-      // baseDetails.features.forEach((featureObj)=> {features.push(featureObj.feature)});
-  // comparedDetails.features.forEach((featureObj)=> {features.push(featureObj.feature)})
-  console.log('all features are: ',features)
+      setBaseStars(data.stars);
+      setComparedStars(compared.stars);
+      data.features.forEach((featureObj) => {
+        featuresSet.add(featureObj.feature)
+        let feat = {};
+        feat[featureObj.feature] = featureObj.value
+        console.log(feat)
+        bFeatures = {...bFeatures, ...feat }
+      });
+      compared.features.forEach((featureObj) => {
+        featuresSet.add(featureObj.feature)
+        let feat = {};
+        feat[featureObj.feature] = featureObj.value
+        cFeatures = {...cFeatures, ...feat }
+      });
+      const set = [...featuresSet]
```

Comment:
QQ - why do you need to use a spread operator and reassign featuresSet before updating state?

Reply:
Sets don't have all of the methods arrays do.  In this case I'm just passing into an array to give it standard array methods.